### PR TITLE
Adding snap for kubefed

### PR DESCRIPTION
**Pull request by ktsakalozos**
Adding kubefed snap

### CODE REVIEW

File: snap/kubefed.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+name: kubefed
+version: 'KUBE_VERSION'
+summary: kubefed controls the Kubernetes cluster federation manager.
+description: |
+  kubefed is a command line interface for running commands against Kubernetes
+  federated clusters.
+grade: stable
+confinement: classic
```
Comment (marcoceppi) [inline snap/kubefed.yaml:8]: Does this need to be classic?

Comment (ktsakalozos) [reply snap/kubefed.yaml:8]: Good question, I just blindly copied the kebectl.yaml.

**Comment by marcoceppi** [comment]
Let's try to make sure strict confinement works. Kubectl and kubelet are
kind of exceptions

File: snap/kubefed.yaml (hunk 1 above)
Comment (ktsakalozos) [reply snap/kubefed.yaml:8]: Did some tests with kubefed on stricted confinement and the user experience is not good.

Kubefed is a cli client that needs to (among others) read and write into .kube/config. The home plug of snaps has the limitation of not being able to access hidden files. Keeping us aligned with best practices means we would better use classic confinement.

**Comment by wwwtyro** [comment]
Merged into upstream PR. Thanks @ktsakalozos!

**Pull request closed without merging**